Packs/cisco-ise/Integrations/cisco-ise/cisco-ise.py

```diff
@@ -661,6 +662,128 @@ def get_blacklist_endpoints():
     return_outputs(tableToMarkdown('CiscoISE Blacklist Endpoints', data, removeNull=True), context, endpoints)
 
 
+def get_endpoint_id_by_name(mac_address=None):
+    """
+    Returns endpoint id by specific mac address
+    Only compatible with Cisco ISE versions 2.3
+    """
+    if not is_mac_address(mac_address):
+        return_error('Given MAC address is invalid')
+
+    api_endpoint = f'/ers/config/endpoint/name/{mac_address}'
+    return http_request('GET', api_endpoint, '')
+
+
+def get_endpoint_id_by_name_command():
+
+    mac_address = demisto.args().get('macAddress')
+
+    if not is_mac_address(mac_address):
+        return_error('Given MAC address is invalid')
+
+    endpoint_data = get_endpoint_id_by_name(mac_address)
+    endpoint_id = endpoint_data.get('ERSEndPoint', {}).get('id', None)
+
+    entry_context = {
+        'Endpoint(val.ID === obj.ID)': {
+            'ID': endpoint_id,
+            'MACAddress': mac_address
+        }
+    }
+
+    return_outputs(f'The endpoint ID is: {endpoint_id}', entry_context, endpoint_id)
+
+
+def get_node_details(name=None):
+    """
+    Returns details for the given Cisco ISE node
+    """
+    if not name:
+        return_error('Given Cisco ISE node is invalid')
+
+    api_endpoint = f'/ers/config/node/name/{name}'
+    return http_request('GET', api_endpoint, '')
+
+
+def get_all_nodes_command():
+    """
+    Returns all nodes in the Cisco ISE Deployment
+    Also sets isLocalIstance in the output
+    """
+
+    instance_ip = urlparse(BASE_URL).netloc
+    try:
+        instance_ip = socket.gethostbyname(instance_ip)
+    except Exception as e:
+        LOG("Failed to get ip address of configured Cisco ISE instance, %s" % (e))
```

Comment:
why LOG and not raise/return_error?
please add inline docstring to further explain

Reply:
Added the raise condition and removed LOG